main.cpp

```diff
@@ -1,30 +1,97 @@
+#define _USE_MATH_DEFINES
+
 #include "main.h"
 #include <cmath>
+#include <math.h>
 #include "rc.h"
+#include "resource.h"
 
 int number_dialog::idd() const {
 	return IDD_NUMBER;
 }
 bool number_dialog::on_init_dialog() {
+	set_int(IDC_EDIT1, number);
 	return true;
 }
 bool number_dialog::on_ok() {
+	try {
+		number = get_int(IDC_EDIT1);
+	}
+	catch (...){
+		return false;
+	}
 	return true;
 }
 
 void main_window::on_paint(HDC hdc) {
+	RECT rc;
+	::GetClientRect(*this, &rc);
+	int r = rc.bottom / 4;
+	HBRUSH brush = CreateSolidBrush(default_color);
+	HGDIOBJ selected_brush = SelectObject(hdc, brush);
+	HPEN pen = CreatePen(PS_SOLID, 1, default_color);
+	HGDIOBJ selected_pen = SelectObject(hdc, pen);
+
+	SetROP2(hdc, R2_NOTXORPEN);
+	if (is_anisotropic) {
+		SetMapMode(hdc, MM_ANISOTROPIC);
+		SetViewportExtEx(hdc, rc.right, rc.bottom, NULL);
+		SetWindowExtEx(hdc, rc.bottom, rc.bottom, NULL);
+//		SetViewportOrgEx(hdc, rc.right / 2, rc.bottom / 2, 0);
+		for (int i = 0; i < default_number; ++i) {
+			double a = 2 * M_PI * i / default_number;
+			double x = rc.bottom / 2 + r * cos(a);
+			double y = rc.bottom / 2 + r * sin(a);
+			::Ellipse(hdc, x + r, y + r, x - r, y - r);
+		}
+
+	}
+	else {
+		for (int i = 0; i < default_number; ++i) {
+			double a = 2 * M_PI * i / default_number;
+			double x = rc.right / 2 + r * cos(a);
+			double y = rc.bottom / 2 + r * sin(a);
+			::Ellipse(hdc, x + r, y + r, x - r, y - r);
+		}
+	}
+
+	DeleteObject(brush);
+	DeleteObject(pen);
 }
 
 void main_window::on_command(int id){
 	switch(id){
-		case ID_COLOR: 
+		case ID_COLOR: {
+			COLORREF custCols[16] = { 0 };

```

Comment:
ovo ne pomaže ako nije _member_ (ili _static_) - svaki puta će biti 16 crnih boja (umjesto zapamćenih "Add to custom colors")